src/main/java/edu/cornell/kfs/concur/service/impl/ConcurEventNotificationServiceImpl.java

```diff
@@ -65,30 +65,40 @@ protected String truncateValidationResultMessageToMaximumDatabaseFieldSize(Strin
     
     @Override
     public void retrieveAndPersistFailedEventQueueReports() {
-        if (shouldProcessFailedEventQueue()) {
+        ConcurFailedEventQueueProcessingController controller = findCOncurFailedEventQueueProcessingController();
+        LOG.info("retrieveAndPersistFailedEventQueueReports, ConcurFailedEventQueueProcessingController name: " + controller.toString());
+        if (controller.queryFailedEventQueue) {
             ConcurEventNotificationListDTO notificationList = getConcurReportsService().retrieveFailedEventQueueNotificationsFromConcur();
-            if (areThereNotificationsToProcess(notificationList)) {
-                LOG.info("retrieveAndPersistFailedEventQueueReports, found " + notificationList.getConcurEventNotificationDTOs().size() + " failed events to process.");
-                for (ConcurEventNotificationDTO concurEventNotificationDTO : notificationList.getConcurEventNotificationDTOs()) {
-                    try {
-                        ConcurEventNotification concurEventNotification = getConcurEventNotificationConversionService().convertConcurEventNotification(concurEventNotificationDTO);
-                        saveConcurEventNotification(concurEventNotification);
-                        getConcurReportsService().deleteFailedEventQueueItemInConcur(findNotificationId(concurEventNotification.getNotificationURI()));
-                    } catch (ParseException e) {
-                        LOG.error("validate():" + e.getMessage(), e);
+            if (controller.logFailedEventQueue) {
+                String notificationListString = getConcurReportsService().retrieveFailedEventQueueNotificationsFromConcurAsString();
```

Comment:
In this setup, the log-only configuration seems to be retrieving the failed event queue twice: Once as a DTO and again as a String. Should the log-only mode avoid doing this both ways? Or do we need the log-only setup to still test the unmarshalling of the XML?

Reply:
I believe it would still be a good idea to test the unmarshalling of the XML, to verify that our XML DTO updates for KFSPTS-18060 will work properly. However, if feasible it would be preferable to do so without having to read the failed event queue twice.

Reply:
I adjusted such that marshalling into the full DTO only happens when we are persististing to the database mode.  As currently configured it will be logged or persisted, not both, so that avoids duplicate calls.

Reply:
Thanks for removing the duplicate service call. Although, given all of the XML DTO namespace changes that will be implemented in a follow-up user story, it would still be helpful to have some way of testing that the DTOs will still get created successfully. Perhaps the CUMarshalService could be used to convert the XML string into the appropriate DTO(s).

If you think it would be better to defer such extra setup to the KFSPTS-18060 user story, that's okay with me, just let me know.

Reply:
I spoke with Chad,  we are going to move the retrieveFailedEventQueueNotificationsFromConcur call back above the logQueue if statement, and then do the update to CUMarshalService in a seperate user story